ingestion/src/metadata/data_quality/validations/table/sqlalchemy/tableDiff.py

```diff
@@ -136,15 +141,81 @@ def _run(self) -> TestCaseResult:
             threshold,
         )
 
+    def get_incomparable_columns(self) -> List[str]:
+        """Get the columns that have incompatible types in the two tables"""
+        table1 = data_diff.connect_to_table(
+            self.runtime_params.table1.serviceUrl,
+            self.runtime_params.table1.path,
+            self.runtime_params.keyColumns,
+            extra_columns=self.runtime_params.extraColumns,
+        ).with_schema()
+        table2 = data_diff.connect_to_table(
+            self.runtime_params.table2.serviceUrl,
+            self.runtime_params.table2.path,
+            self.runtime_params.keyColumns,
+            extra_columns=self.runtime_params.extraColumns,
+        ).with_schema()
+        result = []
+        columns1 = table1.key_columns + table1.extra_columns
+        columns2 = table2.key_columns + table2.extra_columns
+        for col1, col2 in safezip(columns1, columns2):
+            col1_type = self.get_column_type(
+                table1._schema[col1]  # pylint: disable=protected-access
+            )
+            if table2._schema.get(col2) is None:  # pylint: disable=protected-access
+                continue
+            col2_type = self.get_column_type(
+                table2._schema[col2]  # pylint: disable=protected-access
+            )
+
+            if col1_type != col2_type:
+                result.append(col1)
```

Comment:
Can you explain why we choose to return col1 as changed if types are not comparable?

Reply:
I am only returning the column name (which is the same for both tables). So it does not matter which one is returned.

Reply:
changed it to only iterate over columns from the main table. the removed/added is covered by other parts.